src/overlays/ovl_i5/fox_ti.c

```diff
@@ -1,138 +1,5375 @@
+                func_8007A900(temp_ft2, boss->obj.pos.y, temp_ft4, 10.0f, 0xC0, 0x10, 0);
+            }
+        }
+    }
+    boss->dmgType = DMG_NONE;
+}
+
+void func_i5_80193B30(Boss* boss) {
+
+    if (boss->state == boss->swork[1]) {
+        if (boss->swork[1] != 11) {
+            boss->swork[1] = 11;
+        } else {
+            boss->swork[1] = 9;
+        }
+    }
+    switch (boss->swork[1]) {
+        case 8:
+            if ((boss->swork[11] <= 0) && (boss->swork[12] <= 0)) {
+                if (boss->swork[35] == 0) {
+                    boss->swork[1] = 12;
+                } else {
+                    boss->swork[1] = 11;
+                }
+            }
+            break;
+        case 10:
+            if (D_i5_801BBEF0[8] != 0) {
+                boss->swork[1] = 12;
+            }
+            break;
+        case 11:
+            if (boss->swork[35] == 0) {
+                boss->swork[1] = 8;
+            }
+            break;
+    }
+    if (boss->swork[1] != 7) {
+        boss->swork[36]++;
+    } else {
+        boss->swork[36] = 0;
+    }
+    if (boss->swork[36] >= 2) {
+        boss->swork[1] = 7;
+        boss->swork[36] = 0;
+    }
+    if (boss->swork[1] == 8) {
+        boss->swork[3] = 2;
+    } else {
+        boss->swork[3] = 1;
+    }
+    D_i5_801BBEF4[11] = D_i5_801B7738[boss->swork[1]];
+
+    switch (boss->swork[1]) {
+        case 7:
+        case 8:
+        case 9:
+        case 10:
+            if (boss->fwork[35] != 1.0f) {
+                D_i5_801BBEF0[13] = 1;
+                boss->fwork[35] = 1.0f;
+                D_i5_801BBEF4[32] = 10.0f;
+                D_i5_801BBEF4[33] = 10.0f;
+            }
+            break;
+        case 11:
+            boss->fwork[35] = 0.0f;
+            break;
+        case 12:
+            boss->fwork[35] = 0.0f;
+            break;
+    }
+    switch (boss->swork[1]) {
+        case 7:
+            if (boss->swork[9] > 0) {
+                boss->fwork[36] = 1.0f;
+                D_i5_801BBEF0[14] = 1;
+                D_i5_801BBEF4[34] = 1.0f;
+            }
+            if ((boss->swork[10] > 0) && (D_i5_801BBEF0[8] != 0)) {
+                boss->fwork[37] = 1.0f;
+                D_i5_801BBEF0[15] = 1;
+                D_i5_801BBEF4[35] = 1.0f;
+            }
+            boss->swork[34] = 1;
+            break;
+        case 8:
+            boss->swork[34] = 1;
+            break;
+        case 9:
+            boss->swork[34] = 1;
+            break;
+        case 10:
+            boss->swork[34] = 1;
+            break;
+        case 11:
+            boss->swork[34] = 1;
+            boss->swork[24] = 1;
+            boss->fwork[45] = 230.0f;
+            break;
+        case 12:
+            boss->swork[34] = 2;
+            boss->fwork[48] = 0.0f;
+            break;
+        default:
+            break;
+    }
+}
+
+extern s16 D_i5_801BAA50[19];
+extern s16 D_i5_801BAA78[34];
+extern s16 D_i5_801BAABC[4][10];
+extern s16 D_i5_801BAB0C[4][10];
```

Comment:
setting these as const arrays should get them into rodata and let you remove the D symbol pragmas from the bottom.